exercises/two-bucket/.meta/src/reference/java/TwoBucket.java

```diff
@@ -0,0 +1,139 @@
+import java.util.ArrayList;
+
+class TwoBucket {
+	
+    private class State {
+    	int moves;
+    	int bucketOne;
+    	int bucketTwo;
+    	
+    	public State (int moves, int bucketOne, int bucketTwo) {
+    		this.moves = moves;
+    		this.bucketOne = bucketOne;
+    		this.bucketTwo = bucketTwo;
+    	}
+    	
+    	@Override
+    	public boolean equals(Object o) {
+    		State otherState = (State)o;
+    		return this.moves == otherState.moves &&
+    				this.bucketOne == otherState.bucketOne &&
+    				this.bucketTwo == otherState.bucketTwo;
+    	}
+    }
+    
+    private State finalState;
+    
+    int bucketOneCap;
+    int bucketTwoCap;
+    int desiredLiters;
+    String first;
```

Comment:
These should all be `private` since it's good practice to [limit access to variables as much as possible](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html) 🙂 

Also, what's `first` meant to represent? The names of the other variables make it very clear what they are, but this one is a bit more vague. If it's meant to represent the bucket you start with, maybe `firstBucket` or `startBucket` would be more descriptive? 🙂